google/resource_kms_key_ring_test.go

```diff
@@ -0,0 +1,105 @@
+package google
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"log"
+	"os"
+)
+
+func TestAccGoogleKmsKeyRing_basic(t *testing.T) {
+	skipIfEnvNotSet(t,
+		[]string{
+			"GOOGLE_ORG",
+			"GOOGLE_BILLING_ACCOUNT",
+		}...,
+	)
+
+	projectId := "terraform-" + acctest.RandString(10)
+	projectOrg := os.Getenv("GOOGLE_ORG")
+	projectBillingAccount := os.Getenv("GOOGLE_BILLING_ACCOUNT")
+	keyRingName := fmt.Sprintf("tf-test-%s", acctest.RandString(10))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testGoogleKmsKeyRing_recreate(keyRingName),
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testGoogleKmsKeyRing_basic(projectId, projectOrg, projectBillingAccount, keyRingName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGoogleKmsKeyRingExists("google_kms_key_ring.key_ring"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckGoogleKmsKeyRingExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		config := testAccProvider.Meta().(*Config)
+
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Resource not found: %s", resourceName)
+		}
+
+		name := rs.Primary.Attributes["name"]
+		location := rs.Primary.Attributes["location"]
+		project := rs.Primary.Attributes["project"]
+
+		parent := kmsResourceParentString(project, location)
+		keyRingName := kmsResourceParentKeyRingName(project, location, name)
+
+		listKeyRingsResponse, err := config.clientKms.Projects.Locations.KeyRings.List(parent).Do()
```

Comment:
I used list because I don't like it when my tests look similar to the production code.  I thought this would be a good way to still test the intended functionality without making it too similar to `Read`.

I am open to changing it if you would like me to.